jetty9-base/src/main/java/com/google/cloud/runtimes/jetty9/RequestContextScope.java

```diff
@@ -27,18 +27,21 @@
 import java.util.logging.Logger;
 
 /**
- * A Jetty {@link ContextScopeListener} that is called whenever
- * a container managed thread enters or exits the scope of a context and/or request.
- * Used to maintain {@link ThreadLocal} references to the current request and 
- * Google traceID, primarily for logging.
- * @see TracingLogHandler
+ * A Jetty {@link ContextScopeListener} that is called whenever a container managed thread enters or
+ * exits the scope of a context and/or request. Used to maintain {@link ThreadLocal} references to
+ * the current request and Google traceID, primarily for logging.
+ *
+ * @see TraceLoggingEnhancer
  */
 public class RequestContextScope implements ContextHandler.ContextScopeListener {
+
   static final Logger logger = Logger.getLogger(RequestContextScope.class.getName());
 
   private static final String X_CLOUD_TRACE = "x-cloud-trace-context";
   private static final ThreadLocal<Integer> contextDepth = new ThreadLocal<>();
 
+  private final String projectId = System.getProperty("app.deploy.project", "unknown");
```

Comment:
Who will set this system property?
I think you need to use `com.google.cloud.ServiceOptions.getDefaultProjectId()` instead.

Reply:
We have a few tests that do:
```
    datastore = DatastoreOptions.newBuilder()
        .setProjectId(System.getProperty("app.deploy.project"))
        .build()
        .getService();
```
Hence I assumed that property is available... but either way the ServiceOptions approach is much better.

Reply:
That was just designed for testing and it won't work normally. [See readme](https://github.com/GoogleCloudPlatform/jetty-runtime/blob/74a8e127bb7478b1927758a6115b9f8eb775348d/tests/README.md#remote-testing).

`com.google.cloud.ServiceOptions.getDefaultProjectId()` is the way to go.